### PR TITLE
[FLINK-11935][table-planner][table-planner-blink] Remove DateTimeUtils pull-in and fix datetime casting problem

### PR DESCRIPTION
**Pull request by docete**
## What is the purpose of the change
This DateTimeUtils was pulled in in FLINK-7235. And now both legacy planner and blink planner hold a copy of DateTimeUtils which make it very difficult to maintain. On the other hand, Apache calcite keeps carrying forward and fix some bugs. IMO it's time to remove DateTimeUtils pull-in.

The impact to legacy planner:
1) toString of java.sql.Timestamp/Date considers Gregorian cutover which should be ignore. CALCITE-1884 has ignored the cutover, so CAST timestamp/date to STRING should ignore it too.
2) CALCITE-3199 is not merged in avatica-1.15.0 (we currently depends on) which should be fixed by ourself.

The impact to blink planner:
1) SimpleDateFormat(used by casting timestamp to string and vice-versa) considers Gregorian cutover which should be ignore.
2)  CALCITE-3199 is not merged in avatica-1.15.0

## Brief change log

- 8572c6c2 removes DateTimeUtils from legacy planner and blink planner
- cb8a6948 and b5fed1d1 fix legacy planner
- 68acfbf5 and 6c24e992 fix blink planner

## Verifying this change

This change is already covered by existing tests, such as *ScalarFunctionsTest*, *TemporalTypesTest*.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (**yes** / no)
  - If yes, how is the feature documented? (not applicable / docs / **JavaDocs** / not documented)

### CODE REVIEW

File: flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/functions/SqlDateTimeUtils.java [hunk 1]
```diff
@@ -1297,4 +1297,10 @@ private static int parseFraction(String v, int multiplier) {
 		return r;
 	}
 
+	// TODO: remove if CALCITE-3199 fixed
+	//  https://issues.apache.org/jira/browse/CALCITE-3199
+	public static long unixDateCeil(TimeUnitRange range, long date) {
+		return julianDateFloor(range, (int) date + EPOCH_JULIAN, false);
```
Comment (wuchong) [inline flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/functions/SqlDateTimeUtils.java]: Is this correct? It still calls `SqlDateTimeUtils#julianDateFloor` not Calcite's `DateTimeutils`.

Comment (danny0405) [reply flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/functions/SqlDateTimeUtils.java]: Yes, it is, the Julian flag is what CALCITE-3199 fixed.

Comment (docete) [reply flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/functions/SqlDateTimeUtils.java]: the implementation of `Is this correct? It still calls SqlDateTimeUtils#julianDateFloor` is different from `DateTimeutils`, but i didn't find any bad case.

**Comment by docete** [comment]
> Do we have any existing tests to cover the changes in blink planner? For example, it should fail before that cast string to timestamp and date ceil.

already exist.  
date ceil case in ScalarFunctionsTest.testTemporalFloor.  Which we fixed in our copied `DateTimeUtils` in FLINK-13302. And should fix again since we removes it.

for `cast string to timestamp`, it's a little tricky: the original string-> timestamp & timestamp -> string take account of Gregorian cutover and make case passed. No we ignore Gregorian cutover for both path. The case is TemporalTypesTest.testTimePointLiterals (`1500-04-30 12:00:00`)

**Pull request closed without merging**